packages/astro/src/core/compile/compile.ts

```diff
@@ -37,7 +35,7 @@ export async function compile({
 		// use `sourcemap: "both"` so that sourcemap is included in the code
 		// result passed to esbuild, but also available in the catch handler.
 		transformResult = await transform(source, {
-			moduleId,
+			moduleId: filename,
 			pathname: filename,
```

Comment:
I found the `filename` and the `moduleId` passed are always the same (absolute paths or virtual ids). They could only differ in the `resolveId` hook, but since we're compiling in the `transform` hook, it doesn't affect us.

In a future compiler PR, I'll merge `moduleId` and `pathname` options into one.

Reply:
So filename *is* the module ID right? To do head propagation we rely on the moduleId being passed to the compiler to match what is found in the module graph. Just want to make sure that is still going to be happening here.

Reply:
Yup, I tested it and saw it's always identical, even for virtual files. I think the unit test should cover if not as I was able to get it to fail when I pass the wrong id during manual testing.

Reply:
I did another quick review, and I think this part I guess I'm slightly unsure:

https://github.com/withastro/astro/blob/840412128b00a04515156e92c314a929d6b94f6d/packages/astro/src/vite-plugin-astro/index.ts#L244-L245

When you added the `context.modules[0]?.id`, is it because it differs from `context.file`? If I understand the module graph right, [Vite's code](https://github.com/vitejs/vite/blob/896475dc6c7e5f1168e21d556201a61659552617/packages/vite/src/node/server/hmr.ts#L85-L92) should have them equal in most case.

Reply:
I think I did that out of "correctness" rather than a concern about `context.file`. So I think we can go with the change for now and see what happens.